cmd/legacy_main.go

```diff
@@ -268,6 +268,52 @@ func setMaxReadAhead(mountPoint string, readAheadKb int) error {
 	return nil
 }
 
+// setCongestionThreshold sets the fuse settings `congestion-threshold` by writing over
+// `/sys/fs/fuse/connections/<minor>/congestion_threshold`.
+func setCongestionThreshold(mountPoint string, congestionThreshold int) error {
+	_, minor, err := getDeviceMajorMinor(mountPoint)
+	if err != nil {
+		return fmt.Errorf("getting device major/minor for mount point %s: %v", mountPoint, err)
+	}
+
+	sysPathCgTh := filepath.Join("/sys/fs/fuse/connections", fmt.Sprintf("%d", minor), "congestion_threshold")
+	cmd := exec.Command("sudo", "-n", "tee", sysPathCgTh)
+	cmd.Stdin = strings.NewReader(fmt.Sprintf("%d\n", congestionThreshold))
+
+	var stderr strings.Builder
+	cmd.Stderr = &stderr
+	err = cmd.Run()
+	if err != nil {
+		err = fmt.Errorf("setting the congestion threshold %d mount-path %s: %v, stderr: %s", congestionThreshold, mountPoint, err, stderr.String())
+		return err
+	}
+
+	return nil
+}
+
+// setMaxBackground sets the fuse settings `max-background` by writing over
+// `/sys/fs/fuse/connections/<minor>/max_background`.
+func setMaxBackground(mountPoint string, maxBackground int) error {
+	_, minor, err := getDeviceMajorMinor(mountPoint)
+	if err != nil {
+		return fmt.Errorf("getting device major/minor for mount point %s: %v", mountPoint, err)
+	}
+
+	sysPathMxBg := filepath.Join("/sys/fs/fuse/connections", fmt.Sprintf("%d", minor), "max_background")
+	cmd := exec.Command("sudo", "-n", "tee", sysPathMxBg)
+	cmd.Stdin = strings.NewReader(fmt.Sprintf("%d\n", maxBackground))
+
+	var stderr strings.Builder
+	cmd.Stderr = &stderr
+	err = cmd.Run()
+	if err != nil {
+		err = fmt.Errorf("setting the max-background %d on mount-path %s: %v, stderr: %s", maxBackground, mountPoint, err, stderr.String())
```

Comment:
why dont we just log it here. what is the need for constructing an error message, returning and calling doing the logging

Reply:
Following the best practices of method and returned error - so that it can be easily tested.

But given no unit test and used only at one place so logging inside the method only - warning for error and info for success.